### PR TITLE
fix "is" comparison for np.array

### PR DESCRIPTION
**Pull request by raphaelquast**
## PR Summary

fixes incorrect check if numpy array is none.

- `if x` fails for numpy arrays due to array-broadcasting!
- `if x is None` does what it should...

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [x] New features are documented, with examples if plot related.
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -2200,6 +2200,9 @@ def get_cursor_data(self, event):
         if len(info["ind"]) == 1:
             ind, = info["ind"]
             array = self.get_array()
-            return array[ind] if array else None
+            if array is not None:
+                return array[ind]
+            else:
+                return None
```
Comment (timhoffm) [inline lib/matplotlib/collections.py:2206]: Slight preference for keeping the ternary expression:
```suggestion
            return array[ind] if array is not None else None
```

**Review by timhoffm — APPROVED** [review]

**Comment by greglucas** [comment]
It would be nice to get a test for this too, so we don't hit this path again on accident.

I just put up some tests for another PR #22955 in the `contains()` portion of the code: https://github.com/matplotlib/matplotlib/pull/22955/commits/6c22e863236ab5470df2aee19879769a58bf5bf4
A similar test or two could be added here for the `get_cursor_data()` function.

**Comment by QuLogic** [comment]
Feel free to open a followup with the test; I want to merge to get 3.5.2 out.

**Comment by QuLogic** [comment]
meeseeksdev backport to v3.5.x

**Pull request merged**